src/Controller/AdminController.php

```diff
@@ -456,17 +456,44 @@ public function modifyEstimationQuest(
     /**
      * @Route("/anon", name="users_anon")
      * @param UserRepository $users
+     * @param EntityManagerInterface $em
      * @return Response
      * @throws Exception
      */
     // route to anonymise users after 3 years
-    public function anonUsers(UserRepository $users)
+    public function anonUsers(UserRepository $users, EntityManagerInterface $em)
     {
         $date = new DateTime('now');
         $date->sub(new DateInterval('P3Y'));
         $users = $users->findOldUsers($date);
-        return $this->render('bdc/anon.html.twig', [
+
+        $empty = 0; // find users
+        if (empty($users)) {
+            $empty = 1; // find no users
+        } else {
+            // if button is activated, then anonymised users
+            // and give last signin in 1970 to forget them
+            if (isset($_GET["anon"])) {
+                foreach ($users as $user) {
+                    $newEmail = $user->getId()."@olduser.bipbip";
+                    $user->setEmail($newEmail)
+                        ->setLastname("xxx")
+                        ->setPhonenumber("0000000000")
+                        ->setAddress("xxx")
+                        ->setSigninDate(\DateTime::createFromFormat('Y-m-d', "1970-01-01"))
+                        ->setOrganism(null)
+                        ->setCollect(null);
+                    $em->persist($user);
+                    $em->flush();
```

Comment:
get out flush method for win a request bdd

Reply:
It's done, thanks !